src/coreclr/src/jit/emit.cpp

```diff
@@ -5594,7 +5595,17 @@ CORINFO_FIELD_HANDLE emitter::emitFltOrDblConst(double constValue, emitAttr attr
     }
 #endif // TARGET_XARCH
 
-    UNATIVE_OFFSET cnum = emitDataConst(cnsAddr, cnsSize, cnsAlign);
+    UNATIVE_OFFSET cnum;
+    if (attr == EA_16BYTE)
+    {
+        cnum = emitDataConst(cnsAddr, cnsSize, 1);
```

Comment:
`cnsAlign` must be `16` if using the legacy encoding; otherwise it can be any value (which should be either `16` or `1`).